src/IO/ReadBuffer.h

```diff
@@ -197,7 +198,7 @@ class ReadBuffer : public BufferBase
       */
     virtual bool nextImpl() { return false; }
 
-    [[noreturn]] void throwReadAfterEOF()
+    [[noreturn]] static inline void throwReadAfterEOF()
```

Comment:
I doubt about inline.

Reply:
Constructor of Exception can be heavy due to construction of std::string.
If this function is inlined to another small function, that small function may stop to be inlined by itself.

Note: it's just a wild guess.